sycl/source/detail/queue_impl.hpp

```diff
@@ -1040,6 +1047,8 @@ class queue_impl {
   // the fallback implementation of profiling info
   bool MFallbackProfiling = false;
 
+  // Is value presented in MInOrderExternalEvent?
+  std::atomic_bool MHasValueInOrderExternalEvent = false;
```

Comment:
I think the name is confusing, maybe `MInOrderExteranEventIsSet` would be better?

Second, is it correct to assume that we still want the `optional` because default-constructed/empty event is heavy?

Reply:
> Second, is it correct to assume that we still want the optional because default-constructed/empty event is heavy?

Yep, one is heavy. I hope one day we able to implement an empty `sycl::event` with just empty `std::shared_ptr<detail::event_impl> impl`, then optional became unneeded.